src/reactviews/pages/QueryResult/table/plugins/autoColumnSize.plugin.ts

```diff
@@ -0,0 +1,419 @@
+            : width;
+    }
+
+    private getTemplateWidths(
+        rowElements: JQuery[],
+        templates: (JQuery | HTMLElement | string)[],
+    ): number[] {
+        // Write all changes first then read all widths to prevent layout thrashing
+        // (https://developers.google.com/web/fundamentals/performance/rendering/avoid-large-complex-layouts-and-layout-thrashing)
+        const cells: JQuery[] = templates.map((template, index) => {
+            let rowEl = rowElements[index];
+            let cell = jQuery(rowEl.find(".slick-cell"));
+            cell.append(template);
+            jQuery(cell).find("*").css("position", "relative");
+            return cell;
+        });
+
+        return cells.map((cell) => cell.outerWidth()! + 1);
+    }
+
+    private getMaxTextTemplates(
+        allTexts: string[][],
+        columnDefs: Slick.Column<T>[],
+        colIndices: number[],
+        data: Slick.DataProvider<T>,
+        rowElements: JQuery[],
+    ): (JQuery | HTMLElement | string)[] {
+        let numColumns = columnDefs.length;
+        return columnDefs.map((columnDef, index) =>
+            this.getMaxTextTemplate(
+                allTexts[index],
+                columnDef,
+                colIndices[index],
+                data,
+                rowElements[index],
+                numColumns,
+            ),
+        );
+    }
+
+    private getMaxTextTemplate(
+        texts: string[],
+        columnDef: Slick.Column<T>,
+        colIndex: number,
+        data: Slick.DataProvider<T>,
+        rowEl: JQuery,
+        numColumns: number,
+    ): JQuery | HTMLElement | string {
+        let max = 0,
+            maxTemplate: JQuery | HTMLElement | string | undefined;
+        let formatFun = columnDef.formatter;
+        let startTime = Date.now();
+        texts.forEach((text, index) => {
+            let template;
+            if (formatFun) {
+                template = jQuery(
+                    "<span>" +
+                        formatFun(
+                            index,
+                            colIndex,
+                            text,
+                            columnDef,
+                            data.getItem(index),
+                        ) +
+                        "</span>",
+                );
+                text = template.text() || text;
+            }
+            let length = text
+                ? this.getElementWidthUsingCanvas(rowEl, text)
+                : 0;
+            if (length > max) {
+                max = length;
+                maxTemplate = template || text;
+            }
+        });
+        let endTime = Date.now();
+        let timeElapsed = endTime - startTime;
+        let telemetryEvent: WebviewTelemetryActionEvent = {
+            telemetryView: TelemetryViews.QueryResult,
+            telemetryAction: TelemetryActions.AutoColumnSize,
+            additionalMeasurements: {
+                timeElapsed: timeElapsed,
```

Comment:
@cssuh I'd strongly suggest adding units to this value - e.g. `timeElapsedMs`

Reply:
Will do